runelite-client/src/main/java/net/runelite/client/plugins/herbiboars/HerbiboarOverlay.java

```diff
@@ -144,6 +166,17 @@ public Dimension render(Graphics2D graphics)
 			}
 		}
 
+		// Draw herbiboar
+		if (plugin.isHerbiboarRendered())
+		{
+			for (NPC npc : client.getNpcs())
```

Comment:
Would it be possible to save the NPC when setting the var ‘herbiboar rendered’ to true so we can use that instance of NPC instead of looping over all the visible NPCs?

Reply:
No, as the Herbiboar has 3 different states. Also see this:
Herbiboar spawns:
Animation changed: HiscoresOnly: 423
(after some small time:)
Animation changed: Herbiboar: 7687
Spawned: Herbiboar: 7786

Herbiboar gets harvested:
Animation changed: HiscoresOnly: 2277
Animation changed: Herbiboar: 7689

Herbiboar dies:
Animation changed: HiscoresOnly: -1
Animation changed: Herbiboar: 7690

Each time the animation is changed, the herbiboar becomes a different NPC.